### ISSUE TITLE
Ability to add <disp-formula>

### ISSUE THREAD

**Issue by oliver----** [substance/texture#733]
Texture already supports adding `<inline-formula>`. 
Similar thing but as a block-level element.

**Comment by michael** [comment]
We are working on this feature right now. 

@Melissa37 @JGilbert-eLife can you provide samples of math usage in eLife articles? Not completely clear to me yet how we do labelling the equations etc.

We could have automatic numbering, but if that's too hard to do, we can also have the label field editable.

**Comment by eLifeProduction** [comment]
@michael disp-formula elements containing maths may or may not have a label, so I think the second option would be preferable, if possible. inline-formula elements don't have labels.

**Example 1 - with label**
```
<p> ... <disp-formula id="equ1">
        <label>(1)</label>
        <mml:math id="m1">
            <mml:mstyle displaystyle="true" scriptlevel="0">
                <mml:mrow>
                    <mml:msubsup>
                        <mml:mi>σ</mml:mi>
                        <mml:mrow>
                            <mml:mi>R</mml:mi>
                        </mml:mrow>
                        <mml:mrow>
                            <mml:mn>2</mml:mn>
                        </mml:mrow>
                    </mml:msubsup>
                </mml:mrow>
            </mml:mstyle>
        </mml:math>
    </disp-formula>
</p>
```

**Example 2 - without label**
```
<p> ... <disp-formula id="equ2">
        <mml:math id="m2"> ... </mml:math>
    </disp-formula>
</p>
```

**Comment by Integral** [comment]
Is there any reason to put disp-formula inside paragraph?

**Comment by FAtherden-eLife** [comment]
@Integral It's entirely possible that an author wants to continue the paragraph after the formula, for example (actual mml:math removed for readability):
`<p> ... were taken to be<disp-formula id="equ7"><label>(6)</label><mml:math id="m7"></mml:math></disp-formula>where A defines the amplitude of the complex ...`

HTML view:
![screen shot 2018-11-14 at 14 24 25](https://user-images.githubusercontent.com/43879983/48488702-45f04200-e819-11e8-8b9e-21db8e929ae9.png)

In the above, semantically it's correct for the formula to be in a p. 
I can see there might be a case where it might be proper not to capture them in a <p> element but in the vast majority of (eLife) cases the author is referring to them in this fashion, or at the end of a paragraph.

**Comment by FAtherden-eLife** [comment]
No problem. We only use labels in a linear sequence with numbers (and parentheses, as above). Thanks!

**Comment by JGilbert-eLife** [comment]
That's not necessarily true - I think there are examples in our corpus where the authors have used, for example, (A-1), (A-2) etc to distinguish equations in the abstract from the main text, but that;s only happened very infrequently.

**Comment by michael** [comment]
@FAtherden-eLife we would like to treat them as block level elements always. This improves UI/UX and simplifies implementation, as we don't have to consider the inline-block case. We did the same with `<fig>` elements. And we don't allow them inside `<p>`. As a consequence, adding a disp-formula always breaks the paragraph, which I think is valid.

```xml
<p>were taken to be</p>
<disp-formula>...</disp-formula>
<p>whera A ...</p>
```

**Comment by michael** [comment]
Regarding labels, if we can agree on an automatic numbering scheme, this saves you time during authoring/editing. E.g. you can move around formulas and the numbers will update accordingly. If not, you'll have to manually edit them each time.

Is referencing formulas a use-case for eLife? Like an xref that points to a disp-formula?

**Comment by JGilbert-eLife** [comment]
Yes, they can be cited:

`<xref ref-type="disp-formula" rid="equ1">equation 1</xref>`

Regarding the placement of equations, doing it the way you suggest will lead to `<p>` elements that begin with lowercase letters. Not necessarily a problem and I see the argument for doing it your way, but the case where the author wants to continue the 'sentence' started by the display equation, or embed the equation within a sentence as far as the reader is concerned will remain, since this is a standard style of writing.

We get plenty of requests that there is no indentation in the PDF display for the text following an equation, for example, to indicate that the equation is part of a piece of text rather than separate to it.

**Comment by michael** [comment]
If ok, we would like to proceed as described with real block level disp-formulas. And allow referencing them in the main text.

If really needed, we could store some display hint for rendering these special cases, or introduce an inline disp-formula (though this would as said really complicate the UI/UX -> the author needs to choose between a block level or inline formula, this will lead to confusion I'm sure).

**Comment by Melissa37** [comment]
Hi @michael 
Welcome to publishing again! There is a reason for what we've got.

I think the purpose of differentiating disp and inline formula in XML is to allow for our different use cases as described (http://jats.nlm.nih.gov/archiving/tag-library/1.2d1/element/inline-formula.html).

Personally, I think it's going to be too problematic to have new ```<p>```tags starting with lower case letters - we will not be able to rely on that as an indicator for the display purpose of "do not indent this" start of paragraph.

I think you are going to have to deal with disp and inline formula if you want publishers with any math to use Texture.

**Comment by Melissa37** [comment]
Hi @michael this is something we think it really vital for our UX people to be involved with.
How urgent is it for you to work on right now or can you wait a month?

**Comment by michael** [comment]
We are already almost there, I'd suggest we now finish that first iteration of that feature and when ready, demo it to you (and your UX people) to collect missing requirements.

**Comment by oliver----** [comment]
We have implemented basic support for `<disp-formula>` via #1042.

Still open is the requirement which we call now 'paragraph continuation' for which we have opened a new issue: #1052
